### PR TITLE
stdenv: update ARM bootstrap tarballs

### LINKED ISSUE #81972: Unable to build glibc on armv7l
**Describe the bug**

`nix-build -A glibc` doesn't work, either on `master` or `nixos-20.03`. It works on `nixos-19.09` and I have a bisect running at the moment (on some slow hardware, so the results might take some time).

It fails with this log:

```
these derivations will be built:
  /nix/store/aixify1li31zgrzyc00b0hjlqir2j9nk-glibc-2.30.drv
building '/nix/store/aixify1li31zgrzyc00b0hjlqir2j9nk-glibc-2.30.drv'...
unpacking sources
unpacking source archive /nix/store/6kl3xyqvybfv0s6gkhwlxwr0g4k1rwmw-glibc-2.30.tar.xz
source root is glibc-2.30
setting SOURCE_DATE_EPOCH to timestamp 1564633771 of file glibc-2.30/wctype/wctype_l.c
patching sources
applying patch /nix/store/r0rc2prwi70hgw4jray5a8sd41fhvjm7-rpcgen-path.patch
patching file sunrpc/rpc_main.c
applying patch /nix/store/pp29isd0nvy4jrm9q2f92pjlc83xwa0i-nix-locale-archive.patch
patching file locale/loadarchive.c
patching file locale/programs/locale.c
Hunk #1 succeeded at 634 (offset 1 line).
Hunk #2 succeeded at 663 (offset 1 line).
patching file locale/programs/locarchive.c
applying patch /nix/store/5212825a2npxaf7qd81sxfrqq6219rlx-dont-use-system-ld-so-cache.patch
patching file elf/ldconfig.c
patching file elf/Makefile
Hunk #1 succeeded at 589 (offset 30 lines).
patching file sysdeps/generic/dl-cache.h
applying patch /nix/store/mnglr8rr7nl444h7p50ysyq8qd0fm1lm-dont-use-system-ld-so-preload.patch
patching file elf/rtld.c
Hunk #1 succeeded at 1697 (offset 184 lines).
applying patch /nix/store/za0pg7fmysrcwrqcal26fnmzw6vycgdn-fix_path_attribute_in_getconf.patch
patching file sysdeps/unix/confstr.h
applying patch /nix/store/xkd50xxii6k7l1kmw4l5x6xzbhamcs87-allow-kernel-2.6.32.patch
patching file sysdeps/unix/sysv/linux/configure
patching file sysdeps/unix/sysv/linux/configure.ac
patching file sysdeps/unix/sysv/linux/dl-osinfo.h
applying patch /nix/store/nqa9d4yxz1l2cgswzqr1pkm4jfrksm0q-locale-C.diff
patching file localedata/locales/C
patching file inet/idna.c
Hunk #1 succeeded at 27 (offset 2 lines).
configuring
configure flags: --disable-static --prefix=/nix/store/9jznc5gg80v53i6pnj7s0vm1pd5rc27l-glibc-2.30 --bindir=/nix/store/hy49r1awxldbz54zfbx3b1s0r9d4qy57-glibc-2.30-bin/bin --sbindir=/nix/store/hy49r1awxldbz54zfbx3b1s0r9d4qy57-glibc-2.30-bin/sbin --includedir=/nix/store/86qsdcjddw3bj8zq7wjvhpmwdigmm9f5-glibc-2.30-dev/include --oldincludedir=/nix/store/86qsdcjddw3bj8zq7wjvhpmwdigmm9f5-glibc-2.30-dev/include --mandir=/nix/store/hy49r1awxldbz54zfbx3b1s0r9d4qy57-glibc-2.30-bin/share/man --infodir=/nix/store/hy49r1awxldbz54zfbx3b1s0r9d4qy57-glibc-2.30-bin/share/info --docdir=/nix/store/9jznc5gg80v53i6pnj7s0vm1pd5rc27l-glibc-2.30/share/doc/glibc --libdir=/nix/store/9jznc5gg80v53i6pnj7s0vm1pd5rc27l-glibc-2.30/lib --libexecdir=/nix/store/9jznc5gg80v53i6pnj7s0vm1pd5rc27l-glibc-2.30/libexec --localedir=/nix/store/9jznc5gg80v53i6pnj7s0vm1pd5rc27l-glibc-2.30/share/locale -C --enable-add-ons --enable-obsolete-nsl --enable-obsolete-rpc --sysconfdir=/etc --enable-stackguard-randomization --with-headers=/nix/store/3ngfv4pzbr18pxp148zdc5pf65nh0kwf-linux-headers-5.5/include --disable-profile --enable-kernel=3.2.0 --host=arm-linux-gnueabi --build=arm-linux-gnueabi libc_cv_as_needed=no
configure: creating cache config.cache
checking build system type... arm-unknown-linux-gnueabi
checking host system type... arm-unknown-linux-gnueabi
checking for arm-linux-gnueabi-gcc... gcc
checking for suffix of object files... o
checking whether we are using the GNU C compiler... yes
checking whether gcc accepts -g... yes
checking for arm-linux-gnueabi-readelf... readelf
checking whether we are using the GNU C++ compiler... yes
checking whether g++ accepts -g... yes
checking whether g++ can link programs... no
checking for sysdeps preconfigure fragments... aarch64 alpha arm configure: Found compiler is configured for armv7
csky hppa i386 m68k microblaze mips nios2 powerpc riscv s390 sh sparc x86_64
checking for use of fpu sysdeps directories... yes
checking for -fstack-protector... yes
checking for -fstack-protector-strong... yes
checking for -fstack-protector-all... yes
checking for assembler and linker STT_GNU_IFUNC support... yes
checking for gcc attribute ifunc support... no
checking whether the linker supports textrels along with ifunc... yes
checking if compiler warns about alias for function with incompatible types... no
checking sysdep dirs... sysdeps/unix/sysv/linux/arm sysdeps/arm/nptl sysdeps/unix/sysv/linux sysdeps/nptl sysdeps/pthread sysdeps/gnu sysdeps/unix/inet sysdeps/unix/sysv sysdeps/unix/arm sysdeps/unix sysdeps/posix sysdeps/arm/armv7/multiarch sysdeps/arm/armv7 sysdeps/arm/armv6t2 sysdeps/arm/armv6 sysdeps/arm sysdeps/wordsize-32 sysdeps/ieee754/flt-32 sysdeps/ieee754/dbl-64 sysdeps/ieee754 sysdeps/generic
checking for a BSD-compatible install... /nix/store/65sxp86ypi3ca914pq069blazwq9qzr1-bootstrap-tools/bin/install -c
checking whether ln -s works... yes
checking whether /nix/store/hszwh50hga8i4gx09r85w14x038wg0pf-bootstrap-stage2-gcc-wrapper-/bin/as is GNU as... yes
checking whether /nix/store/hszwh50hga8i4gx09r85w14x038wg0pf-bootstrap-stage2-gcc-wrapper-/bin/ld is GNU ld... yes
checking for /nix/store/hszwh50hga8i4gx09r85w14x038wg0pf-bootstrap-stage2-gcc-wrapper-/bin/as... /nix/store/hszwh50hga8i4gx09r85w14x038wg0pf-bootstrap-stage2-gcc-wrapper-/bin/as
checking version of /nix/store/hszwh50hga8i4gx09r85w14x038wg0pf-bootstrap-stage2-gcc-wrapper-/bin/as... 2.31.1, ok
checking for /nix/store/hszwh50hga8i4gx09r85w14x038wg0pf-bootstrap-stage2-gcc-wrapper-/bin/ld... /nix/store/hszwh50hga8i4gx09r85w14x038wg0pf-bootstrap-stage2-gcc-wrapper-/bin/ld
checking version of /nix/store/hszwh50hga8i4gx09r85w14x038wg0pf-bootstrap-stage2-gcc-wrapper-/bin/ld... 2.31.1, ok
checking for gnumake... no
checking for gmake... no
checking for make... make
checking version of make... 4.2.1, ok
checking for gnumsgfmt... no
checking for gmsgfmt... no
checking for msgfmt... no
checking for makeinfo... no
checking for sed... sed
checking version of sed... 4.4, ok
checking for gawk... gawk
checking version of gawk... 4.1.3, ok
checking for bison... bison
checking version of bison... 3.5.1, ok
checking if gcc is sufficient to build libc... no
checking for arm-linux-gnueabi-nm... nm
checking for python3... python3
checking version of python3... 3.7.6, ok
configure: error:
*** These critical programs are missing or too old: compiler
*** Check the INSTALL file for required versions.
builder for '/nix/store/aixify1li31zgrzyc00b0hjlqir2j9nk-glibc-2.30.drv' failed with exit code 1
```

**To Reproduce**

`nix-build -A glibc`

**Metadata**
`nix-shell -p nix-info --run "nix-info -m"` 

```
 - system: `"armv7l-linux"`
 - host os: `Linux 4.19.75, NixOS, 19.09.git.2b23e13 (Loris)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.3.1`
 - nixpkgs: `/home/j/src/nixpkgs2`
```

Maintainer information:
```yaml
# a list of nixpkgs attributes affected by the problem
attribute: 
  - glibc
  - "pretty much everything lol"
```

### PR DESCRIPTION
**Pull request by grahamc**
###### Motivation for this change

follow-up to https://github.com/NixOS/nixpkgs/pull/79793

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by grahamc** [comment]
ehh....

```
hash mismatch in fixed-output derivation '/nix/store/crk7rqpclwd6c282lkzjpakk76iv8p7y-7a7a0266d4687eb1b44496d4d185b9ffa9b1c6260c5df1c40a4999c612312bbd':
  wanted: sha256:1g9b649cd6a91b2g2p8c4v3b3agzp62x3m4n8jsb2zk8sik04yks
  got:    sha256:18qc6w2yykh7nvhjklsqb2zb3fjh4p9r22nvmgj32jr1mjflcsjn
```

**Comment by thefloweringash** [comment]
The `executable` flag in fetchurl changes the hash algorithm to `recursive` to include the executable metadata. see [fetchurl/default.nix][].

[fetchurl/default.nix]: https://github.com/NixOS/nixpkgs/blob/d7a68e9bce3ceb302aa12e4339cce119575cbd33/pkgs/build-support/fetchurl/default.nix#L137

File: pkgs/stdenv/linux/bootstrap-files/armv5tel.nix [hunk 1]
```diff
@@ -1,12 +1,17 @@
 {
+  # Note: do not use Hydra as a source URL. Ask a member of the
+  # infrastructure team to mirror the job.
   busybox = import <nix/fetchurl.nix> {
-    url = https://hydra.nixos.org/build/112609163/download/2/busybox;
-    sha256 = "0dc5471dc6a5f69ad98eb7445f51a61e88aa5792d7a677025bf012bdb513b763";
+    # from job: https://hydra.nixos.org/job/nixpkgs/cross-trunk/bootstrapTools.armv5tel.dist/latest
+    # from build: https://hydra.nixos.org/build/113692702
```
Comment (thefloweringash) [inline pkgs/stdenv/linux/bootstrap-files/armv5tel.nix]: The build url in the comment is the same for all three architectures, and points to an armv7l build.

**Comment by thefloweringash** [comment]
The name of the `busybox` derivation appears to be significant. If I run this after fixing the hash, I get
```
$ nix-build . -A hello --argstr system armv7l-linux
[ ... ]
building '/nix/store/cff954wzdw9r66kwi3gipihmw8rb1ffs-bootstrap-tools.drv' on 'ssh://alex@alex7'...
crk7rqpclwd6c282lkzjpakk76iv8p7y-7a7a0266d4687eb1b44496d4d185b9ffa9b1c6260c5df1c40a4999c612312bbd: applet not found
builder for '/nix/store/cff954wzdw9r66kwi3gipihmw8rb1ffs-bootstrap-tools.drv' failed with exit code 127
```

Adding `name = "busybox"` is enough to make it start building.

**Comment by thefloweringash** [comment]
My suggested changes:
```diff
diff --git a/pkgs/stdenv/linux/bootstrap-files/armv5tel.nix b/pkgs/stdenv/linux/bootstrap-files/armv5tel.nix
index e8da070c397..18a94fb1b40 100644
--- a/pkgs/stdenv/linux/bootstrap-files/armv5tel.nix
+++ b/pkgs/stdenv/linux/bootstrap-files/armv5tel.nix
@@ -3,14 +3,15 @@
   # infrastructure team to mirror the job.
   busybox = import <nix/fetchurl.nix> {
     # from job: https://hydra.nixos.org/job/nixpkgs/cross-trunk/bootstrapTools.armv5tel.dist/latest
-    # from build: https://hydra.nixos.org/build/113692702
+    # from build: https://hydra.nixos.org/build/114081916
+    name = "busybox";
     url = https://tarballs.nixos.org/sha256/3091d8e22778d023d51c50526ac4911b617db38b7f092fa3332c1d3ff5bd53b7;
-    sha256 = "3091d8e22778d023d51c50526ac4911b617db38b7f092fa3332c1d3ff5bd53b7";
+    sha256 = "0qxp2fsvs4phbc17g9npj9bsm20ylr8myi5pivcrmxm5qqflgi8d";
     executable = true;
   };
   bootstrapTools = import <nix/fetchurl.nix> {
     # from job: https://hydra.nixos.org/job/nixpkgs/cross-trunk/bootstrapTools.armv5tel.dist/latest
-    # from build: https://hydra.nixos.org/build/113692702
+    # from build: https://hydra.nixos.org/build/114081916
     url = https://tarballs.nixos.org/sha256/16c307ce6d22166f2da506744caf90fcb529e3cbd003da57bcd8c1c3812ebfd7;
     sha256 = "16c307ce6d22166f2da506744caf90fcb529e3cbd003da57bcd8c1c3812ebfd7";
   };
diff --git a/pkgs/stdenv/linux/bootstrap-files/armv6l.nix b/pkgs/stdenv/linux/bootstrap-files/armv6l.nix
index 8541091ded1..eac1fa32b90 100644
--- a/pkgs/stdenv/linux/bootstrap-files/armv6l.nix
+++ b/pkgs/stdenv/linux/bootstrap-files/armv6l.nix
@@ -3,14 +3,15 @@
   # infrastructure team to mirror the job.
   busybox = import <nix/fetchurl.nix> {
     # from job: https://hydra.nixos.org/job/nixpkgs/cross-trunk/bootstrapTools.armv6l.dist/latest
-    # from build: https://hydra.nixos.org/build/113692702
+    # from build: https://hydra.nixos.org/build/114081910
+    name = "busybox";
     url = https://tarballs.nixos.org/sha256/d96c7656bf43e9ab24949a0b7f6c7c4a4daf1f89ce10ea6e86aa645cbff65bce;
-    sha256 = "d96c7656bf43e9ab24949a0b7f6c7c4a4daf1f89ce10ea6e86aa645cbff65bce";
+    sha256 = "1q02537cq56wlaxbz3s3kj5vmh6jbm27jhvga6b4m4jycz5sxxp6";
     executable = true;
   };
   bootstrapTools = import <nix/fetchurl.nix> {
     # from job: https://hydra.nixos.org/job/nixpkgs/cross-trunk/bootstrapTools.armv6l.dist/latest
-    # from build: https://hydra.nixos.org/build/113692702
+    # from build: https://hydra.nixos.org/build/114081910
     url = https://tarballs.nixos.org/sha256/787ce488369504c29c57e36dd25eaf84e8cc8423d9f3da7359daa0023aaafb99;
     sha256 = "787ce488369504c29c57e36dd25eaf84e8cc8423d9f3da7359daa0023aaafb99";
   };
diff --git a/pkgs/stdenv/linux/bootstrap-files/armv7l.nix b/pkgs/stdenv/linux/bootstrap-files/armv7l.nix
index c788b1e436c..be69dd4ca69 100644
--- a/pkgs/stdenv/linux/bootstrap-files/armv7l.nix
+++ b/pkgs/stdenv/linux/bootstrap-files/armv7l.nix
@@ -3,14 +3,15 @@
   # infrastructure team to mirror the job.
   busybox = import <nix/fetchurl.nix> {
     # from job: https://hydra.nixos.org/job/nixpkgs/cross-trunk/bootstrapTools.armv7l.dist/latest
-    # from build: https://hydra.nixos.org/build/113692702
+    # from build: https://hydra.nixos.org/build/114081942
+    name = "busybox";
     url = https://tarballs.nixos.org/sha256/7a7a0266d4687eb1b44496d4d185b9ffa9b1c6260c5df1c40a4999c612312bbd;
-    sha256 = "7a7a0266d4687eb1b44496d4d185b9ffa9b1c6260c5df1c40a4999c612312bbd";
+    sha256 = "18qc6w2yykh7nvhjklsqb2zb3fjh4p9r22nvmgj32jr1mjflcsjn";
     executable = true;
   };
   bootstrapTools = import <nix/fetchurl.nix> {
     # from job: https://hydra.nixos.org/job/nixpkgs/cross-trunk/bootstrapTools.armv7l.dist/latest
-    # from build: https://hydra.nixos.org/build/113692702
+    # from build: https://hydra.nixos.org/build/114081942
     url = https://tarballs.nixos.org/sha256/c727c21ebb863f6e8e88cfef73f4ddb534ae942df1a0f9d2e1d1825b6f843a86;
     sha256 = "c727c21ebb863f6e8e88cfef73f4ddb534ae942df1a0f9d2e1d1825b6f843a86";
   };
```

**Comment by lopsided98** [comment]
I applied @thefloweringash's changes and rebased on staging (to get the unpack-boostrap-tools fix), and I am building stdenv for armv7l and armv6l to make sure this still works.

**Comment by thefloweringash** [comment]
I've successfully built up to `hello` for armv7l. I don't think you need to rebase, since 05fc2a35240d4bdae638e273b56a2ff2722a40a9 is an ancestor of this commit.

**Comment by lopsided98** [comment]
Oh I see; I assumed staging hadn't been merged yet. I'll build it for armv6l then.

File: pkgs/stdenv/linux/bootstrap-files/armv5tel.nix (hunk 1 above)
Comment (grahamc) [reply pkgs/stdenv/linux/bootstrap-files/armv5tel.nix]: hah... uh .... :)

**Comment by thefloweringash** [comment]
I've verified the tarballs contents match their hash and correspond to the named builds. ✅ 

Without `name = "busybox"` I'm getting the "applet not found" error.

**Comment by vcunat** [comment]
PR #80497 was merged directly to master, not staging.  It's not a mass rebuild.

**Comment by vcunat** [comment]
Is the URL style intentionally different from the other stdenv-bootstrap tarballs?  Those have e.g.
```
http://tarballs.nixos.org/stdenv-linux/x86_64/c5aabb0d603e2c1ea05f5a93b3be82437f5ebf31/bootstrap-tools.tar.xz;
```
(where the hash is for nixpkgs git commit, not for contents)

**Comment by grahamc** [comment]
Not different on purpose, but I'm feeling too tired today to follow the (new-to-me) directions properly. Can you give me specific instructions? We need to do that before we merge this.

**Comment by vcunat** [comment]
I have no idea about reasons for the style... so your `hashedMirrors` style might even be preferable AFAIK.  (It's also never garbage-collected, right?)  When I was (involved in) updating the tarballs because of the last glibc bump, I just discovered how the URL was composed and kept to the style.

**If** we were to follow the style currently used for x86_64, I believe the URLs would be:
```
http://tarballs.nixos.org/stdenv-linux/armv5tel/0eb0ddc4dbe3cd5415c6b6e657538eb809fc3778/busybox
http://tarballs.nixos.org/stdenv-linux/armv5tel/0eb0ddc4dbe3cd5415c6b6e657538eb809fc3778/bootstrap-tools.tar.xz
http://tarballs.nixos.org/stdenv-linux/armv6el/0eb0ddc4dbe3cd5415c6b6e657538eb809fc3778/busybox
http://tarballs.nixos.org/stdenv-linux/armv6el/0eb0ddc4dbe3cd5415c6b6e657538eb809fc3778/bootstrap-tools.tar.xz
http://tarballs.nixos.org/stdenv-linux/armv7el/0eb0ddc4dbe3cd5415c6b6e657538eb809fc3778/busybox
http://tarballs.nixos.org/stdenv-linux/armv7el/0eb0ddc4dbe3cd5415c6b6e657538eb809fc3778/bootstrap-tools.tar.xz
```

_I'm not sure why we typically use http://tarballs.nixos.org. I'd expect http**s**:// to be preferable nowadays, but I don't feel strongly about this case._

**Comment by lopsided98** [comment]
`hello` built successfully for armv6l.

**Comment by vcunat** [comment]
I see there's now a Hydra machine and [a jobset](https://hydra.nixos.org/jobset/nixpkgs/nixpkgs-unstable-armv7l) for `armv7l` (!)   That verifies bootstrap in there, but apparently much was failed due to out-of-memory killer.

**Comment by misuzu** [comment]
I have successfully built `pkgs.nix` using this branch on top of master on my armv7 device so it's seems to be good to go.

**Comment by grahamc** [comment]
Updated to use the new URLs from @vcunat (thank you for the hard-to-mistake instructions :).)

File: pkgs/stdenv/linux/bootstrap-files/armv5tel.nix [hunk 2]
```diff
@@ -1,12 +1,18 @@
 {
+  # Note: do not use Hydra as a source URL. Ask a member of the
+  # infrastructure team to mirror the job.
   busybox = import <nix/fetchurl.nix> {
-    url = https://hydra.nixos.org/build/112609163/download/2/busybox;
-    sha256 = "0dc5471dc6a5f69ad98eb7445f51a61e88aa5792d7a677025bf012bdb513b763";
+    # from job: https://hydra.nixos.org/job/nixpkgs/cross-trunk/bootstrapTools.armv5tel.dist/latest
+    # from build: https://hydra.nixos.org/build/114203025
+    url = http://tarballs.nixos.org/stdenv-linux/armv5tel/0eb0ddc4dbe3cd5415c6b6e657538eb809fc3778/busybox;
```
Comment (cole-h) [inline pkgs/stdenv/linux/bootstrap-files/armv5tel.nix]:
```suggestion
    url = "http://tarballs.nixos.org/stdenv-linux/armv5tel/0eb0ddc4dbe3cd5415c6b6e657538eb809fc3778/busybox";
```

File: pkgs/stdenv/linux/bootstrap-files/armv5tel.nix [hunk 3]
```diff
@@ -1,12 +1,18 @@
...
+    sha256 = "0qxp2fsvs4phbc17g9npj9bsm20ylr8myi5pivcrmxm5qqflgi8d";
     executable = true;
   };
-
   bootstrapTools = import <nix/fetchurl.nix> {
-    url = https://hydra.nixos.org/build/112609163/download/1/bootstrap-tools.tar.xz;
-    sha256 = "ca0564eca4eb944649ce10ec70859640427bf2241243af62812b163176487e02";
+    # from job: https://hydra.nixos.org/job/nixpkgs/cross-trunk/bootstrapTools.armv5tel.dist/latest
+    # from build: https://hydra.nixos.org/build/114203025
+    url = http://tarballs.nixos.org/stdenv-linux/armv5tel/0eb0ddc4dbe3cd5415c6b6e657538eb809fc3778/bootstrap-tools.tar.xz;
```
Comment (cole-h) [inline pkgs/stdenv/linux/bootstrap-files/armv5tel.nix]:
```suggestion
    url = "http://tarballs.nixos.org/stdenv-linux/armv5tel/0eb0ddc4dbe3cd5415c6b6e657538eb809fc3778/bootstrap-tools.tar.xz";
```

File: pkgs/stdenv/linux/bootstrap-files/armv6l.nix [hunk 4]
```diff
@@ -1,12 +1,18 @@
 {
+  # Note: do not use Hydra as a source URL. Ask a member of the
+  # infrastructure team to mirror the job.
   busybox = import <nix/fetchurl.nix> {
-    url = https://hydra.nixos.org/build/112609441/download/2/busybox;
-    sha256 = "e6f6aecb675e924a96516f4379445dd2c0ba8b9c438fbfbaa2dc14ccce2802e0";
+    # from job: https://hydra.nixos.org/job/nixpkgs/cross-trunk/bootstrapTools.armv6l.dist/latest
+    # from build: https://hydra.nixos.org/build/114202834
+    url = http://tarballs.nixos.org/stdenv-linux/armv6l/0eb0ddc4dbe3cd5415c6b6e657538eb809fc3778/busybox;
```
Comment (cole-h) [inline pkgs/stdenv/linux/bootstrap-files/armv6l.nix]:
```suggestion
    url = "http://tarballs.nixos.org/stdenv-linux/armv6l/0eb0ddc4dbe3cd5415c6b6e657538eb809fc3778/busybox";
```

File: pkgs/stdenv/linux/bootstrap-files/armv6l.nix [hunk 5]
```diff
@@ -1,12 +1,18 @@
...
+    sha256 = "1q02537cq56wlaxbz3s3kj5vmh6jbm27jhvga6b4m4jycz5sxxp6";
     executable = true;
   };
-
   bootstrapTools = import <nix/fetchurl.nix> {
-    url = https://hydra.nixos.org/build/112609441/download/1/bootstrap-tools.tar.xz;
-    sha256 = "7a3f20def1a17ebf0edb5a92c403558429bcc2ac3d931b5e1bd88606cb217778";
+    # from job: https://hydra.nixos.org/job/nixpkgs/cross-trunk/bootstrapTools.armv6l.dist/latest
+    # from build: https://hydra.nixos.org/build/114202834
+    url = http://tarballs.nixos.org/stdenv-linux/armv6l/0eb0ddc4dbe3cd5415c6b6e657538eb809fc3778/bootstrap-tools.tar.xz;
```
Comment (cole-h) [inline pkgs/stdenv/linux/bootstrap-files/armv6l.nix]:
```suggestion
    url = "http://tarballs.nixos.org/stdenv-linux/armv6l/0eb0ddc4dbe3cd5415c6b6e657538eb809fc3778/bootstrap-tools.tar.xz";
```

File: pkgs/stdenv/linux/bootstrap-files/armv7l.nix [hunk 6]
```diff
@@ -1,12 +1,18 @@
 {
+  # Note: do not use Hydra as a source URL. Ask a member of the
+  # infrastructure team to mirror the job.
   busybox = import <nix/fetchurl.nix> {
-    url = https://hydra.nixos.org/build/112609103/download/2/busybox;
-    sha256 = "566a469dac214b31e4abdb0a91d32550bab1be5858d329e1b6074eef05370ca3";
+    # from job: https://hydra.nixos.org/job/nixpkgs/cross-trunk/bootstrapTools.armv7l.dist/latest
+    # from build: https://hydra.nixos.org/build/114203060
+    url = http://tarballs.nixos.org/stdenv-linux/armv7l/0eb0ddc4dbe3cd5415c6b6e657538eb809fc3778/busybox;
```
Comment (cole-h) [inline pkgs/stdenv/linux/bootstrap-files/armv7l.nix]:
```suggestion
    url = "http://tarballs.nixos.org/stdenv-linux/armv7l/0eb0ddc4dbe3cd5415c6b6e657538eb809fc3778/busybox";
```

File: pkgs/stdenv/linux/bootstrap-files/armv7l.nix [hunk 7]
```diff
@@ -1,12 +1,18 @@
...
+    sha256 = "18qc6w2yykh7nvhjklsqb2zb3fjh4p9r22nvmgj32jr1mjflcsjn";
     executable = true;
   };
-
   bootstrapTools = import <nix/fetchurl.nix> {
-    url = https://hydra.nixos.org/build/112609103/download/1/bootstrap-tools.tar.xz;
-    sha256 = "79fa2d7722aeb856c7c9b62a3fd64b6d261fd6f6bcbac486f0a2a7d823210550";
+    # from job: https://hydra.nixos.org/job/nixpkgs/cross-trunk/bootstrapTools.armv7l.dist/latest
+    # from build: https://hydra.nixos.org/build/114203060
+    url = http://tarballs.nixos.org/stdenv-linux/armv7l/0eb0ddc4dbe3cd5415c6b6e657538eb809fc3778/bootstrap-tools.tar.xz;
```
Comment (cole-h) [inline pkgs/stdenv/linux/bootstrap-files/armv7l.nix]:
```suggestion
    url = "http://tarballs.nixos.org/stdenv-linux/armv7l/0eb0ddc4dbe3cd5415c6b6e657538eb809fc3778/bootstrap-tools.tar.xz";
```

**Review by cole-h — COMMENTED** [review]
I'm still new here, so maybe there's a valid reason you're not following [RFC 45](https://github.com/NixOS/rfcs/pull/45). If there is, feel free to disregard this review (though I would like to hear an explanation so I can keep it in mind for the future).

EDIT: LGTM
image

**Comment by grahamc** [comment]
I need reviews on two things:

First, validate that for each URL, the string between `stdenv-linux/` and `/0eb0ddc4dbe3cd5415c6b6e657538eb809fc3778` matches the name of the `.nix` file it is contained. Example: `stdenv-linux/armv5tel/0eb0ddc4dbe3cd5415c6b6e657538eb809fc3778` is in `armv5tel.nix`.

Second, clone the repo and: `nix-build pkgs/stdenv/linux/bootstrap-files/armv{5tel,6l,7l}.nix` then `nix-build pkgs/stdenv/linux/bootstrap-files/armv{5tel,6l,7l}.nix --check`

**Comment by misuzu** [comment]
I can confirm that URL's is correct.

After running `nix-build pkgs/stdenv/linux/bootstrap-files/armv{5tel,6l,7l}.nix`:
```
% nix-build pkgs/stdenv/linux/bootstrap-files/armv{5tel,6l,7l}.nix --check
checking outputs of '/nix/store/4pqc9wnk7bl58ldsrp19bag989apycyk-bootstrap-tools.tar.xz.drv'...
checking outputs of '/nix/store/5s7769ikm3q8xj9lqnimxmykmv5skzd1-bootstrap-tools.tar.xz.drv'...
warning: rewriting hashes in '/nix/store/07z7gmwq8hp426lz3jqffkgab2jxi90l-bootstrap-tools.tar.xz'; cross fingers
checking outputs of '/nix/store/df6rmkjx6ibmnciidb1xjb8cvd2s6vsa-bootstrap-tools.tar.xz.drv'...
warning: rewriting hashes in '/nix/store/3m5jx9hphpvgyk1y4xl6briydlr79ry4-bootstrap-tools.tar.xz'; cross fingers
checking outputs of '/nix/store/3f0x83pm1czzzh3xa53sh4k6hvhsrk1r-busybox.drv'...
warning: rewriting hashes in '/nix/store/rs65pfdrqvzw6h65i2jwdxkksvp0m459-busybox'; cross fingers
checking outputs of '/nix/store/cn5zp12vp0k2bm97k6ykr84mkw43bvlh-busybox.drv'...
warning: rewriting hashes in '/nix/store/mg8nzxs7mmg6ff47fgv8i61g6m818r8g-busybox'; cross fingers
checking outputs of '/nix/store/cw8afs6r5dnbvqwf7bnc8w86pd4x756b-busybox.drv'...
warning: rewriting hashes in '/nix/store/imsbql435j383ha6q6nlwn5bmqdnaqhq-busybox'; cross fingers
warning: rewriting hashes in '/nix/store/iwd9w34aif01zrs1bxz5kcdkawkv7i0a-bootstrap-tools.tar.xz'; cross fingers
/nix/store/07z7gmwq8hp426lz3jqffkgab2jxi90l-bootstrap-tools.tar.xz
/nix/store/rs65pfdrqvzw6h65i2jwdxkksvp0m459-busybox
/nix/store/3m5jx9hphpvgyk1y4xl6briydlr79ry4-bootstrap-tools.tar.xz
/nix/store/mg8nzxs7mmg6ff47fgv8i61g6m818r8g-busybox
/nix/store/iwd9w34aif01zrs1bxz5kcdkawkv7i0a-bootstrap-tools.tar.xz
/nix/store/imsbql435j383ha6q6nlwn5bmqdnaqhq-busybox
```

**Comment by vcunat** [comment]
I believe this MR should be picked to `release-20.03`.  Any opinions?  @NixOS/nixos-release-managers  _New glibc is in there as well, so the bootstrap is surely broken in there._

**Comment by lopsided98** [comment]
If this is backported, make sure #79793 is as well.

**Comment by expipiplus1** [comment]
It would be great to have this backported

**Pull request merged**